packages/gateway/src/metric/emitter.ts

```diff
@@ -0,0 +1,93 @@
+import { EventEmitter } from 'events';
+import { Pipeline } from 'ioredis';
+import { prefixLogger } from '@graphql-portal/logger';
+import { config } from '@graphql-portal/config';
+import redisConnect from '../redis/connect';
+import { MeshPubSub, ResolverData } from '@graphql-mesh/types';
+import { serializer, transformResolverData } from './utils';
+import MetricsChannels from './channels.enum';
+
+const logger = prefixLogger('metric-emitter');
+export const metricEmitter = new EventEmitter();
+
+const subscribe = async (redisConnectionString: string, pubsub: MeshPubSub): Promise<void> => {
+  const redis = await redisConnect(redisConnectionString);
```

Comment:
too much redisConnects, please change the subscriber in redis/index to be the singleton and reuse the single connection